mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -105,6 +112,21 @@ public Mailbox findMailboxByPath(MailboxPath path) throws MailboxException {
         }
     }
 
+    private Mailbox keepOnlyOneMailbox(ResultSet resultSet) {
+        ImmutableList<SimpleMailbox> mailboxes = CassandraUtils.convertToStream(resultSet)
+            .map(this::mailbox)
+            .sorted(Comparator.comparing(mailbox -> mailbox.getMailboxId().serialize()))
+            .collect(Guavate.toImmutableList());
+        Mailbox mailbox = mailboxes.get(0);
```

Comment:
this code is awful : there's plenty of better API to do that, skip vs subList, getFirst vs get(0), etc